src/Table/TableController.cc

```diff
@@ -185,7 +183,7 @@ void TableController::OnFilterRequest(
 
 void TableController::OnFileListRequest(
     const CARTA::CatalogListRequest& file_list_request, CARTA::CatalogListResponse& file_list_response) {
-    fs::path root_path(_top_level_folder);
+    fs::path root_path(ProgramSettings::GetInstance().top_level_folder);
```

Comment:
I suggest adding a `_settings` property to this class as well (same rationale as for the other classes).